doc/topics/history.rst

```diff
@@ -0,0 +1,161 @@
+Zope-based applications.  A major difficulty was helping people deploy
+the applications in their own environments, which varied not just
+between customers, but also between customer machines. The customer
+environments, including operating system versions, libraries and
+Python modules installed were not well defined and subject to change
+over time.
+
+We realized that we needed to insulate ourselves from the customer
+environments [#ultimately-not-enough]_ to have any chance of
+predictable success.
+
+We decided to provide our own Python builds into which we
+installed the application.  These were automated with `make
+<https://en.wikipedia.org/wiki/Make_(software)>`_.  Customers would
+receive `tar <https://en.wikipedia.org/wiki/Tar_(computing)>`_ files,
+expand them and run ``make``.  We referred to these as "build outs".
+
+Python
+======
+
+Later, as the applications we were building became more complex, some
+of us wanted to be able to use Python, rather than make, to automate
+deployments.  In 2005, we created an internal prototype that used
+builds defined using `ConfigParser-formatted configuration files
+<https://docs.python.org/2/library/configparser.html>`_.  File
+sections described things to be built and there were a few built-in
+build recipes and eventually facilities for implementing custom
+recipes.
+
+By this time, we were hosting most of the applications we were
+building, but we were still building Python and critical libraries
+ourselves as part of deployment, to isolate ourselves from system
+Python and library packages.
+
+After several months of successful experience with the prototype, we
+decided to build what became zc.buildout based on our experience,
+making a recipe framework a main idea.
+
+.. _buildout_and_packaging:
+
+Buildout and packaging
+======================
+
+Around this time, `setuptools and easy_install
+<https://en.wikipedia.org/wiki/Setuptools>`_ were released, providing
+automated download and installation of Python packages *and their
+dependencies*.  Because we built large applications, this was
+something we'd wanted for some time and had even begun building a
+package manager ourselves.  Part of the rationale for creating a new
+Buildout version, beyond the initial prototype, was to take advantage of
+the additional automation that setuptools promised.
+
+Initially, we tried to leverage the ``easy_install`` command
+[#easy_install_module]_, but the goals of ``easy_install`` and
+Buildout were at odds.  ``easy_install`` sought to make it easy for
+humans to install and upgrade packages manually, in an ad hoc manner.
+While it installed dependencies, it didn't upgrade them.  It didn't
+provide ways of managing an installation as a whole.  Buildout, on the
+other hand, was all about automation and repeatability.
+
+To achieve Buildout's goals, it was necessary to interact with
+setuptools at a much lower level and to write quite a bit of packaging
```

Comment:
quite a bit *more* packaging logic

*more* is missing, *of* no longer needed when added.